vortex-array/src/array/bool/compute/compare.rs

```diff
@@ -0,0 +1,115 @@
+use std::ops::{BitAnd, BitOr, BitXor, Not};
+
+use vortex_error::VortexResult;
+use vortex_expr::operators::Operator;
+
+use crate::array::bool::BoolArray;
+use crate::compute::compare::CompareArraysFn;
+use crate::{Array, ArrayTrait, IntoArray};
+
+impl CompareArraysFn for BoolArray {
+    fn compare_arrays(&self, other: &Array, op: Operator) -> VortexResult<Array> {
+        let flattened = other.clone().flatten_bool()?;
+        let lhs = self.boolean_buffer();
+        let rhs = flattened.boolean_buffer();
+        let result_buf = match op {
+            Operator::EqualTo => lhs.bitxor(&rhs).not(),
+            Operator::NotEqualTo => lhs.bitxor(&rhs),
+
+            Operator::GreaterThan => lhs.bitand(&rhs).bitxor(&lhs),
+            Operator::GreaterThanOrEqualTo => {
+                let gt = lhs.bitand(&rhs).bitxor(&lhs);
+                let eq = &lhs.bitxor(&rhs).not();
+                gt.bitor(eq)
+            }
+            Operator::LessThan => lhs.bitor(&rhs).bitxor(&lhs),
+            Operator::LessThanOrEqualTo => {
+                let eq = lhs.bitxor(&rhs).not();
+                let lt = lhs.bitor(&rhs).bitxor(&lhs);
+                lt.bitor(&eq)
+            }
+        };
+        let present_buf = self
+            .validity()
+            .to_logical(self.len())
+            .to_present_null_buffer()?
```

Comment:
You should use regular null buffer, not present null buffer, since you can shortcut